### PR TITLE
Added Dorico Support

### PR DESCRIPTION
**Pull request by EvilArchitech**

### CODE REVIEW

File: music21/configure.py [hunk 1]
```diff
@@ -1596,7 +1607,7 @@ def run(self, forceList=None):
 _DOC_ORDER = []
 
 
-class TestUserInput(unittest.TestCase):  # pragma: no cover
+class TestExternal(unittest.TestCase):  # pragma: no cover
```
Comment (jacobtylerwalls) [inline music21/configure.py:1610]: This is not named TestExternal so that it won't run on CI, where we would have to patch the interactive user inputs so that the job doesn't hang.

File: music21/configure.py [hunk 2]
```diff
@@ -1762,15 +1773,15 @@ def run():
     else:
         # only if running tests
         t = Test()
-        te = TestUserInput()
+        te = TestExternal()
```
Comment (jacobtylerwalls) [inline music21/configure.py:1776]: sim.

File: music21/configure.py [hunk 3]
```diff
@@ -1762,15 +1773,15 @@ def run():
...
 
         if len(sys.argv) < 2 or sys.argv[1] in ['all', 'test']:
             import music21
             music21.mainTest(Test)
 
         # arg[1] is test to launch
         elif sys.argv[1] == 'te':
-            # run test user input
+            # run test external
```
Comment (jacobtylerwalls) [inline music21/configure.py:1784]: I suppose this shouldn't be `'te'` anymore.

File: music21/configure.py [hunk 4]
```diff
@@ -44,11 +44,13 @@
 reFinaleReaderApp = re.compile(r'Finale Reader\.app', IGNORECASE)
 reMuseScoreApp = re.compile(r'MuseScore.*\.app', IGNORECASE)
 reMuseScoreExe = re.compile(r'Musescore.*\\bin\\MuseScore.*\.exe', IGNORECASE)
+reDoricoApp = re.compile(r'Dorico.*\.app', IGNORECASE)
+reDoricoExe = re.compile(r'Dorico.*\.exe', IGNORECASE)
 
-urlMusic21 = 'https://web.mit.edu/music21'
+urlMusic21 = 'http://web.mit.edu/music21'
```
Comment (jacobtylerwalls) [inline music21/configure.py:50]: I'm curious why this unrelated change? Sphinx `linkcheck` will emit errors for redirects from http -> https, so we just link to https.

File: music21/configure.py [hunk 5]
```diff
@@ -44,11 +44,13 @@
...
 urlMuseScore = 'http://musescore.org'
-urlGettingStarted = 'https://web.mit.edu/music21/doc/'
-urlMusic21List = 'https://groups.google.com/g/music21list'
+urlGettingStarted = 'http://web.mit.edu/music21/doc/'
+urlMusic21List = 'http://groups.google.com/group/music21list'
```
Comment (jacobtylerwalls) [inline music21/configure.py:53]: sim., I think Sphinx will complain if we make a change.

File: music21/configure.py [hunk 6]
```diff
@@ -1479,8 +1490,8 @@ def getDialogs(self):
 
     def _introduction(self):
         msg = []
-        msg.append('Welcome to the music21 Configuration Assistant. You will be guided '
-                   + 'through a number of questions to install and set up music21. '
+        msg.append('Welcome the music21 Configuration Assistant. You will be guided '
+                   + 'through a number of questions to install and setup music21. '
```
Comment (jacobtylerwalls) [inline music21/configure.py:1494]: I'm curious why this unrelated change? Also "set up" is the correct verb phrase here; "setup" is a noun.

**Review by jacobtylerwalls — CHANGES_REQUESTED** [review]
Thanks for moving this forward! I'll approve the CI run once you can push changes -- I'd like to be sure the job won't hang because of the change to `TestUserInput`.

File: music21/configure.py (hunk 6 above)
Comment (EvilArchitech) [reply music21/configure.py:1494]: I did not (intentionally) make any such change.

File: music21/configure.py (hunk 4 above)
Comment (EvilArchitech) [reply music21/configure.py:50]: I did not (Intentionally) make any such change.

File: music21/configure.py (hunk 1 above)
Comment (EvilArchitech) [reply music21/configure.py:1610]: I don't understand the comment.

File: music21/configure.py (hunk 3 above)
Comment (EvilArchitech) [reply music21/configure.py:1784]: Sorry, I don't understand the comment.

File: music21/configure.py (hunk 2 above)
Comment (EvilArchitech) [reply music21/configure.py:1776]: sim, I don't understand the comment. :)

**Comment by jacobtylerwalls** [comment]
@EvilArchitech Hey, I think I figured out what happened here. It's likely you began by editing an outdated version of this file, perhaps from version 6.7, and then uploaded it to GitHub. Instead you'll need to base your changes on an updated version of music21 based on the master branch. You can download it from GitHub and re-upload it if you're more comfortable with uploading than with doing a `git clone`.

**Comment by EvilArchitech** [comment]
Got it, I will take a look at the master.  I did the upload only because I have an SSO linked corporate login to GitHub, and it was a pain to switch logins without messing up other repos in MS Code.

**Comment by mscuthbert** [comment]
I think it'd be easier to close the PR and then open a new one from when you're sure you're on the current music21 master.

File: music21/configure.py (hunk 1 above)
Comment (mscuthbert) [reply music21/configure.py:1610]: The class must be called TestExternal.

**Comment by mscuthbert** [comment]
Closing so there's no danger of merging.  Feel free to open a new PR based on latest master.
